test_scripts/Defects/4_6/1389_SDL_responds_to_ListFiles_without_list_of_existed_files_in_app_folder_after_re_registration.lua

```diff
@@ -0,0 +1,81 @@
+---------------------------------------------------------------------------------------------------
+-- User story: https://github.com/SmartDeviceLink/sdl_core/issues/1389
+--
+-- Precondition:
+-- 1) Set proper value to smartDeviceLink.ini for property:
+-- 2) AppStorageFolder = storage
+-- 3) Allow PutFile RPC for FULL HMILevel in policy table.
+-- Description:
+-- SDL responds to ListFiles without list of existed files in app folder after re-registration
+-- Steps to reproduce:
+-- 1) Start SDL and HMI
+-- 2) Connect mobile and create mobile session
+-- 3) Register mobile App
+-- 4) Activate mobile App
+-- 5) Send RPC "PutFile" from mobile App
+-- 6) Unregister mobile App
+-- 7) Register mobile App
+-- 8) Send RPC "ListFiles" from mobile App
+-- Expected result:
+-- SDL responds to mobile app with list of existed files in app folder after re-registration.
+-- Actual result:
+-- SDL responds to ListFiles without list of existed files in app folder after re-registration.
+---------------------------------------------------------------------------------------------------
+--[[ Required Shared libraries ]]
+local runner = require('user_modules/script_runner')
+local common = require('test_scripts/Defects/commonDefects')
+
+--[[ Local Variables ]]
+local putFileParams = {
+	requestParams = {
+		syncFileName = "icon.png",
+		fileType = "GRAPHIC_PNG"
+	},
+	filePath = "files/icon.png"
+}
+
+local requestParams = {}
+
+local responseParams = {
+	success = true,
+	resultCode = "SUCCESS"
+}
+
+local allParams = {
+	requestParams = requestParams,
+	responseParams = responseParams
+}
+
+--[[ Local Functions ]]
+local function PutFile(self)
+	local cid = self.mobileSession1:SendRPC("PutFile", putFileParams.requestParams, putFileParams.filePath)
+	self.mobileSession1:ExpectResponse(cid, {success = true, resultCode = "SUCCESS"})
+end
+
+local function ListFiles(params, self)
+	local cid = self.mobileSession1:SendRPC("ListFiles", params.requestParams)
+	self.mobileSession1:ExpectResponse(cid, params.responseParams )
+	:ValidIf (function(_, data)
```

Comment:
According to defect description in ListFiles response must be checked the list of files in filenames.
According to sequence filenames = { "icon.png" }